### DISCUSSION TITLE
does AudioFileReader support .flac files?

### DISCUSSION THREAD

**Discussion by yueyinqiu** [naudio/NAudio#1130] [Q&A] [answered]
Hello! I'm trying to read some `.flac` files with NAudio but I've met some problems.

I think the following code could describe it more clear:

```csharp
using NAudio.Wave;

var path = "p225_001_mic2.flac";

using var reader = new AudioFileReader(path);
Console.WriteLine(reader.Length / sizeof(float)); // 96000

var buffer = new float[1];
for(int i = 0; ; i++)
{
    var b = reader.Read(buffer, 0, 1);
    if (b < 1)
    {
        Console.WriteLine(i); // 98473
        break;
    }
}
```

And this is the audio file:

[p225_001_mic2.zip](https://github.com/naudio/NAudio/files/14472079/p225_001_mic2.zip)

I assume that `reader.Length / sizeof(float)` should be equal to the final value of `i`. Actually It works well when it's a .wav file, but it doesn't seem to work on .flac files.

Meanwhile, the following codes just print `-1`:

```csharp
using NAudio.Wave;

var path = "p225_001_mic2.flac";

using var reader = new AudioFileReader(path);
Console.WriteLine(reader.ReadByte()); // -1
```

So have I missed something? Or perhaps `AudioFileReader` actually does not support flac files at all (but I have seen someone else are using NAudio to play .flac files, like #1072)?

**Comment by markheath** [comment, marked as answer]
There are two ways to play FLAC in NAudio. You can first of all use `MediaFoundationReader` and rely on the FLAC codec supplied with Windows. This should work fine although some people have reported bugs with the Windows FLAC support.
The other option is that now NAudio comes with `SoundFilePlayer` which uses `libsndfile` which can also play FLAC, although you will need to supply the binary yourself ([see docs here](https://github.com/naudio/NAudio/blob/main/Docs/CrossPlatformAudioFilesWithSoundFile.md))
